src/CGAL_Nef_polyhedron.cc

```diff
@@ -23,25 +25,28 @@ CGAL_Nef_polyhedron CGAL_Nef_polyhedron::operator+(const CGAL_Nef_polyhedron &ot
 
 CGAL_Nef_polyhedron& CGAL_Nef_polyhedron::operator+=(const CGAL_Nef_polyhedron &other)
 {
-	(*this->p3) += (*other.p3);
+	this->p3.reset(new CGAL_Nef_polyhedron3((*this->p3) + (*other.p3)));
 	return *this;
 }
 
 CGAL_Nef_polyhedron& CGAL_Nef_polyhedron::operator*=(const CGAL_Nef_polyhedron &other)
 {
-	(*this->p3) *= (*other.p3);
+	this->p3.reset(new CGAL_Nef_polyhedron3((*this->p3) * (*other.p3)));
 	return *this;
 }
 
 CGAL_Nef_polyhedron& CGAL_Nef_polyhedron::operator-=(const CGAL_Nef_polyhedron &other)
 {
-	(*this->p3) -= (*other.p3);
+	this->p3.reset(new CGAL_Nef_polyhedron3((*this->p3) - (*other.p3)));
 	return *this;
 }
 
 CGAL_Nef_polyhedron &CGAL_Nef_polyhedron::minkowski(const CGAL_Nef_polyhedron &other)
 {
-	(*this->p3) = CGAL::minkowski_sum_3(*this->p3, *other.p3);
+	this->p3.reset(new CGAL_Nef_polyhedron3(CGAL::minkowski_sum_3(
+		const_cast<CGAL_Nef_polyhedron3&>(*this->p3),
+		const_cast<CGAL_Nef_polyhedron3&>(*other.p3)
+	)));
```

Comment:
> Postcondition
>     If either of the input polyhedra is non-convex, it is modified during the computation, i.e., it is decomposed into convex pieces.
> 
from https://doc.cgal.org/latest/Minkowski_sum_3/group__PkgMinkowskiSum3Ref.html

There are some other places where I had to `const_cast` which seemed like mistakes in CGAL's api (e.g. see `CGAL_Nef_polyhedron3::bytes()` below), but in this case I'm not sure if a full copy should be made of each opearand (might make 3D minkowski even slower).

I have no idea if this potential decomposition could have an adverse effect on subsequent operations, in cases where a modified input object has been already cached, then re-used elsewhere in the script.

Reply:
Yes, I'm suspicious of this specific const_cast, although I think the others are safe.  I think this one is formal UB because of the modifications, and thus ambiguously hazardous beyond the cache issues.  The copy should not be a significant performance difference on a minkowski.